clar2wasm/src/words/equal.rs

```diff
@@ -161,146 +166,153 @@ impl ComplexWord for IndexOf {
-                                .local_get(end_offset);
-                            // STACK: [offset, end_offset]
-
-                            else_.binop(BinaryOp::I32GeU).if_else(
-                                InstrSeqType::new(
-                                    &mut generator.module.types,
-                                    &[],
-                                    &[ValType::I32, ValType::I64, ValType::I64],
-                                ),
-                                |then| {
-                                    // Reached the end of the sequence
-                                    // and not found the element.
-                                    then.i32_const(0).local_get(index).i64_const(0);
-                                    // STACK: [0, index_lo, index_hi]
-                                },
-                                |else_| {
-                                    // Increment index by 1
-                                    // and continue loop.
-                                    else_
-                                        .local_get(index)
-                                        .i64_const(1)
-                                        .binop(BinaryOp::I64Add)
-                                        .local_set(index)
-                                        .br(loop_id);
-                                },
-                            );
-                        },
-                    );
-                });
-            },
-        );
+                        )
+                    }
+                    SequenceElementType::Byte => {
+                        // The element type is a byte, so we can just push the
+                        // offset and size = 1 to the stack.
+                        let size = 1;
+                        loop_.local_get(offset).i32_const(size);
+                        // STACK: [offset, size]
+
+                        (size, generator.save_to_locals(loop_, &item_ty, true))
+                        // STACK: []
+                    }
+                    SequenceElementType::UnicodeScalar => {
+                        // The element type is a unicode scalar, so we can just push the
+                        // offset and size = 4 to the stack.
+                        let size = 4;
+                        loop_.local_get(offset).i32_const(size);
+                        // STACK: [offset, size]
+
+                        (size, generator.save_to_locals(loop_, &item_ty, true))
+                        // STACK: []
+                    }
+                };
+
+                // Check item and element equality.
+                // And push the result of the comparison onto the top of the stack.
+                let _res = wasm_equal(
+                    &item_ty,
+                    &item_ty,
+                    generator,
+                    loop_,
+                    &item_locals,
+                    &elem_locals,
+                );
```

Comment:
Should deal with this error here, it has been ignored in the original algorithm apparently.

Reply:
_res wasn't being used. Maybe that's why it was overlooked. Handled it